### PR TITLE
add git functions

### PR DESCRIPTION
**Pull request by beasknees**
I used ChatGPT to generate these since I don't know python. The logic seemed to make since. Working on testing it now but want to open the PR to get feedback.

### CODE REVIEW

File: tiller_python_service/git_actions.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+def git_commit(repo_path, message):
+    try:
+        repo = Repo(repo_path)
+        repo.index.commit(message)
+        return "Commit successful with message: {}".format(message)
+    except GitCommandError as e:
+        return str(e)
+
+def git_push(repo_path, branch='master'):
```
Comment (beasknees) [inline tiller_python_service/git_actions.py:48]: test

File: tiller_python_service/git_actions.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+def git_add_all(repo_path):
+    try:
+        repo = Repo(repo_path)
+        repo.git.add(A=True)  # Adds all changes
+        return "All files added successfully"
+    except GitCommandError as e:
+        return str(e)
+
+def git_commit(repo_path, message):
```
Comment (beasknees) [inline tiller_python_service/git_actions.py:40]: testing

**Comment by beasknees** [comment]
> No need to deal with any virtualenv setup, if you just stand the service up with `docker-compose up` it'll take care of everything you need in the container.

Oops. I dropped the pipenv stuff and forgot to update the comment.

File: tiller_python_service/git_actions.py [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    # Create a temporary directory.
+    target_dir = tempfile.mkdtemp(dir="/tmp")
+
+    # Clone the repository.
+    repo_name = url.split("/")[-1].replace(".git", "")
+    full_repo_path = os.path.join(target_dir, repo_name)
+
+    result = subprocess.run(["git", "clone", url, full_repo_path])
+
+    if result.returncode != 0:
```
Comment (AnotherOctopus) [inline tiller_python_service/git_actions.py:26]: hello

**Pull request merged**